src/micro_http/src/common/headers.rs

```diff
@@ -286,6 +312,174 @@ impl Headers {
     }
 }
 
+/// Wrapper over supported AcceptEncoding.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Encoding {
+    /// Encoding: gzip.
+    Gzip,
+    /// Encoding: compress.
+    Compress,
+    /// Encoding: deflate.
+    Deflate,
+    /// Encoding: Brotli.
+    Br,
+    /// Encoding: Identity(no encoding).
+    Identity,
+    /// Encoding: Wildcard(any encoding that was not already specified).
+    WildCard,
+}
+
+impl Default for Encoding {
+    /// Default value and the only one accepted so far is no encoding(i.e identity)
+    fn default() -> Self {
+        Self::Identity
+    }
+}
+
+impl Encoding {
+    /// Parses a byte slice into a Encoding structure for a HTTP request. Encoding
+    /// must be ASCII, so also UTF-8 valid.
+    ///
+    /// # Errors
+    /// The function returns `InvalidRequest` when the byte stream is empty, `InvalidValue` when the
+    /// byte stream contains a logical error, `UnsupportedValue` when the byte stream contains a
+    /// value that is not supported yet and `InvalidUtf8String` when the byte stream contains
+    /// invalid characters.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use micro_http::Encoding;
+    ///
+    /// assert!(Encoding::try_from(b"deflate").is_ok());
+    /// assert!(Encoding::try_from(b"indentity;q=0").is_err());
+    /// ```
+    pub fn try_from(bytes: &[u8]) -> Result<Self, RequestError> {
+        if bytes.is_empty() {
+            return Err(RequestError::InvalidRequest);
+        }
+        match std::str::from_utf8(bytes) {
+            Ok(headers_str) => {
+                let mut q_value_identity = false;
+                let entry = headers_str.split(',').collect::<Vec<&str>>();
+                for encoding in entry {
+                    let aux = encoding.split(';').collect::<Vec<&str>>();
+                    match aux[0].trim() {
+                        // The client wants encoding to be gzip.
+                        "gzip" => Ok(Self::Gzip),
+                        // The client wants encoding to be compress.
+                        "compress" => Ok(Self::Compress),
+                        // The client wants encoding to be deflate.
+                        "deflate" => Ok(Self::Deflate),
+                        // The client wants encoding to be brotli.
+                        "br" => Ok(Self::Br),
+                        // The client wants no encoding.
+                        "identity" => {
```

Comment:
This branch is quite complex. Maybe it would be worth it separating it in another method.